server/app/Models/Doc.php

```diff
@@ -230,67 +220,45 @@ public function getPagesAttribute()
         return $this->getPages();
     }
 
-    public function getCommentCount()
-    {
-        return $this->comments()->count();
-    }
-
     public function getCommentCountAttribute()
     {
-        return $this->getCommentCount();
+        // note that this is only the top level comment count
+        return $this->comments
+            ->filter(function ($item) {
+                return !$item->isNote();
+            })
+            ->count()
+            ;
```

Comment:
I'm ok with raw SQL messiness if avoids doing in PHP things the database should be doing.  The database is always going to be better at these things than the webserver; if PHP runs out of memory it's just going to die and not recover for that request.  Anyway, yes, this differentiation may only be temporary so it might be overkill - I just want to make sure we're using good patterns.